LaTeX: add `tectonic` (#1803)
JuliaDocs/Documenter.jl: docs/src/man/other-formats.md

  highlighter [Pygments](https://pygments.org/) installed.
* You need the [_DejaVu Sans_ and _DejaVu Sans Mono_](https://dejavu-fonts.github.io/) fonts installed.

### Compiling using Tectonic 

The documentation can be also built using the

> [Tectonic](https://tectonic-typesetting.github.io) typesetting system. It is required to have a `tectonic`

Reviewer: ```suggestion
[Tectonic](https://tectonic-typesetting.github.io) LaTeX engine. It is required to have a `tectonic`
```

> executable present in `PATH`, or to provide it using the `tectonic` keyword:

Reviewer: ```suggestion
available in `PATH`, or to provide a path to the binary using the `tectonic` keyword:
```

```
using Documenter

# Executable `tectonic` is present in `PATH`
makedocs(
    format = Documenter.LaTeX(platform="tectonic"), 
    ...)

> # The path to `tectonic` is provided by the JLL

Reviewer: ```suggestion
# The path to `tectonic` is provided by the tectonic_jll package
```

> import tectonic_jll: tectonic

Reviewer: ```suggestion
using tectonic_jll: tectonic
```
